cnf-certification-test/lifecycle/podrecreation/podrecreation.go

```diff
@@ -65,7 +71,7 @@ func CordonHelper(name, operation string) error {
 	return retryErr
 }
 
-func CountPodsWithDelete(nodeName string, isDelete bool) (count int, err error) {
+func CountPodsWithDelete(nodeName, mode string) (count int, err error) {
```

Comment:
I believe this function is doing too many things, which ends up with a bit convoluted code and some code for UT only (DeleteBackground mode). I suggest breaking it down into two different but simpler functions to be called separately in the test case:
```
GetPodsToDeleteFromNode(node string) ([]*v1.Pods, error)
DeletePods([]v1.Pods, watchForDeletion bool) error
```
As this is just a refactor suggestion, this could be done in further PRs, though.

Reply:
ok will refactor this in another PR